sos/report/plugins/__init__.py

```diff
@@ -2662,6 +2662,37 @@ def get_process_pids(self, process):
                 continue
         return pids
 
+    def filter_namespaces(self, namespaces, ns_pattern = None, ns_max = 0):
+        """Filter a list of namespaces by regex pattern or max number of
+        namespaces (options originally present in the networking plugin.)
+        """
+        out_ns = []
+
+        # Regex initialization outside of for loop
+        if ns_pattern:
+            pattern = (
+                '(?:%s$)' % '$|'.join(ns_pattern.split()).replace('*', '.*')
+                )
+        for ns in namespaces:
+            # if ns_pattern defined, append only namespaces
+            # matching with pattern
+            if ns_pattern:
+                if bool(re.match(pattern, ns)):
+                    out_ns.append(ns)
```

Comment:
This means, if *both* `ns_pattern` and `ns_max` is set, the later option is ignored? I am not sure how broad scenario it is to call networking plugin with both limits applied, but still I see it as a legitimate one.

Could this scenario be supported as well?

Reply:
Hi Pavel,

Thanks for reviewing.

Good point. I wondered about that as well, as to why the options are mutually exclusive, but their combination does seem  a legitimate use case, as you mentioned. (e.g., all namespaces matching a regex, but at most N, in case the system has way too many.)

I guess it can be supported, but I'd have to defer to more experienced folks/author/maintainers, as I'm fresh new to sosreport code, as you may have seen from previous suggestions. :)

For now/this PR, I focused more on keeping the current behavior (sure enough) and just changing places for current things; but would be happy to take that route / add another commit with your suggestion, if that is the consensus.

cheers,
Mauricio

Reply:
Ah right, now I recall there was some discussion in #1916 without a consensus. Let dont change the behaviour and just copy the current one, here.